### ISSUE TITLE
1.0: DO Node Pods Not Cleaned Up

### ISSUE THREAD

**Issue by nelsam** [supergiant/supergiant#368]
I have supergiant running on DO, and it regularly creates/deletes nodes on demand.  This is really cool!  What's not quite so cool is the 9 pages of `kube-proxy-{kube-name}-node-{node-id}`.  It seems like these should get cleaned up when the nodes are removed, but they're left hanging around instead.

**Comment by nelsam** [comment]
That was with 0.15.6.  I'm trying with latest master, now.

**Comment by jordanrinke** [comment]
Try logging in and out of the UI - we do have a random bug in which an event will trigger over and over in the UI that duplicates data in the display only. Let us know if that changes anything or if persist. It is only the kube-proxy that is doing that?

**Comment by nelsam** [comment]
It definitely persists - I found the bug after logging in fresh to check on
the cluster today. I had nine pages of pods, most of them proxies.

I also had two nodes in DO that were no longer known to supergiant.

I'll let you know if the same thing happens with current master.

**Comment by nelsam** [comment]
Latest master appears to have the same problem with pods.  The cluster had a master with id `wb6vk` and a node with id `rpa68` when the following screenshot was taken:

![supergiant](https://user-images.githubusercontent.com/1019908/31727957-3fa362d0-b3e8-11e7-9374-c645f069f2f7.jpg)

You can see that there are proxies left around for other node IDs in that list.

**Comment by jordanrinke** [comment]
Depending how your setup is running, you should have a log file being written either in the same directory as the SG API, or wherever you set it to run from. If you can post that log up here, or browse through it yourself - it should indicate when it is creating those kube-proxy pods, and any failures it might be having in removing it. A copy of the log, or your sanitized personal findings would be helpful. You can PM it to me on Slack as well if it has any semi-sensitive data in it.

**Comment by nelsam** [comment]
I'm seeing this show up in the logs with debug log level turned on:
```
time="2017-10-18T15:47:12Z" level=error msg="Recovered in RecurringService core.NodeObserver: runtime error: invalid memory address or nil pointer dereference"
time="2017-10-18T15:47:12Z" level=debug msg="goroutine 228 [running]:\nruntime/debug.Stack(0xc420326fc0, 0xc4207319c8, 0x2)\n\t/usr/lib64/go/src/runtime/debug/stack.go:24 +0xa7\ngithub.com/supergiant/supergiant/pkg/core.(*RecurringService).recover(0xc42033ca20)\n\t/home/valczir/dev/supergiant/src/github.com/supergiant/supergiant/pkg/core/recurring_service.go:55 +0x199\npanic(0xdaacc0, 0x16bf150)\n\t/usr/lib64/go/src/runtime/panic.go:491 +0x283\ngithub.com/supergiant/supergiant/pkg/core.(*NodeObserver).Perform(0xc42000e3d0, 0xc420211ef8, 0xc42004cc90)\n\t/home/valczir/dev/supergiant/src/github.com/supergiant/supergiant/pkg/core/node_observer.go:87 +0xb21\ngithub.com/supergiant/supergiant/pkg/core.(*RecurringService).perform(0xc42033ca20, 0xf3e3d8, 0xc42033ca20)\n\t/home/valczir/dev/supergiant/src/github.com/supergiant/supergiant/pkg/core/recurring_service.go:41 +0x5f\ngithub.com/supergiant/supergiant/pkg/core.(*RecurringService).tick(0xc42033ca20)\n\t/home/valczir/dev/supergiant/src/github.com/supergiant/supergiant/pkg/core/recurring_service.go:32 +0x6d\ngithub.com/supergiant/supergiant/pkg/core.(*RecurringService).Run(0xc42033ca20)\n\t/home/valczir/dev/supergiant/src/github.com/supergiant/supergiant/pkg/core/recurring_service.go:24 +0x2f\ncreated by github.com/supergiant/supergiant/pkg/core.(*Core).InitializeBackground\n\t/home/valczir/dev/supergiant/src/github.com/supergiant/supergiant/pkg/core/core.go:251 +0x103\n"
```
Everything  else looks normal thus far, but I only just switched to debug level.

**Comment by jordanrinke** [comment]
Interesting, I will have to spin up a test instance against DO to figure out what object isn't being setup correctly but fair warning it likely won't happen in the next few days.

**Comment by nelsam** [comment]
Here's the other interesting bit from those logs:
```
time="2017-10-18T16:19:17Z" level=info msg="Capacity service is creating node with size 512mb"
time="2017-10-18T16:19:17Z" level=debug msg="PUT :: Begin  : provisioning Node 67c14e91-c428-41e0-b7c0-cf7549998a2f"
time="2017-10-18T16:19:17Z" level=error msg="Error in RecurringService : Timed out waiting for Helm cmd 'list'"
time="2017-10-18T16:19:17Z" level=info msg="Terminating node "
time="2017-10-18T16:19:17Z" level=debug msg="PUT :: Begin  : deleting Node 67c14e91-c428-41e0-b7c0-cf7549998a2f"
time="2017-10-18T16:19:17Z" level=warning msg="Deleting Node 5 which has no provider_id"
time="2017-10-18T16:19:17Z" level=debug msg="DEL :: End    : deleting Node 67c14e91-c428-41e0-b7c0-cf7549998a2f"
time="2017-10-18T16:19:19Z" level=warning msg="Failed to tag Droplet 66871546 with value dragonfit-node-c59bt"
time="2017-10-18T16:19:20Z" level=debug msg="DEL :: End    : provisioning Node 67c14e91-c428-41e0-b7c0-cf7549998a2f"
```
I'm going to remove the cluster for now and fall back on a more manual k8s setup.  Let me know if you need any help debugging this, outside of normal 9-5 (mountain time) weekday hours.

**Comment by gopherstein** [comment]
This has been addressed in 2.0. This is in beta and will be released by Dec 15th
